code/go/0chain.net/miner/protocol_receive_integration_tests.go

```diff
@@ -308,6 +310,28 @@ func getNotarisationInfo(not *Notarization) *cases.NotarisationInfo {
 	}
 }
 
+// HandleNotarizedBlockMessage - handles a notarized block for a previous round.
+func (mc *Chain) HandleNotarizedBlockMessage(ctx context.Context,
+	msg *BlockMessage) {
+	mc.handleNotarizedBlockMessage(ctx, msg)
+}
+
+// handleVRFShare - handles the vrf share.
+func (mc *Chain) HandleVRFShare(ctx context.Context, msg *BlockMessage) {
+	state := crpc.Client().State()
+
+	if state.RoundHasFinalizedConfig != nil && state.RoundHasFinalizedConfig.Round+1 == int(msg.VRFShare.Round) && chain.IsSpamReceiver(state, msg.VRFShare.Round-1) {
```

Comment:
The finalized round is usually 2 or 3 rounds behind the current round, basically if round 10 is finalized, the next round 11 must have been notarized(guaranteed by protocol). So perhaps name it as RoundHasNotarizedConfig would be more make sense.